### ISSUE TITLE
CMake errors in chainbase, abieos, appbase andn fc

### ISSUE THREAD

**Issue by cc32d9** [EOSIO/eos#6381]
I'm working on a project that will read `state_history_plugin` data, decode and pass for further processing:
https://github.com/EOSChronicleProject/eos-chronicle

I'm trying to reuse EOSIO libraries as much as possible, and there are several issues, documented in https://github.com/EOSChronicleProject/eos-chronicle/issues/1

OS: Ubuntu 18.10, with all prerequisites, and `fill-postgresql` compiles fine.

`chainbase`, `abieos`, `appbase` require modifications in CMakeLists.txt as described in my link.

`fc` depends on GMP library, but adding `fc` as subdirectory fails because cmake cannot find the rules for GMP.

**Comment by spoonincode** [comment]
Basically what this issue is getting at is that we need to fix the cmake files for the following libraries so that they function correctly outside of being included by eos (make them more proper cmake files)

- [ ] chainbase
- [ ] appbase
- [ ] abieos
- [ ] fc

**Comment by spoonincode** [comment]
Rolling back the changes I made and removing me from assignee. Doing this correctly across all platforms we support was becoming too much of a time sink at the moment.

**Comment by cc32d9** [comment]
I made it work finally:
https://github.com/EOSChronicleProject/eos-chronicle/commit/ab71ae001e718f1b3d0403176306f7a1659ba169#diff-af3b638bc2a3e6c650974192a53c7291

It would be good to move library-specific dependencies, such as `GNUInstallDirs` into each library's Cmake files, instead of figuring them out in the main program cmake rules.

Also `FindBoost.cmake` that comes with FC seems to be broken. I had to add the FC modules to the list after cmake foud Boost using the system libraries.
